query-engine/connectors/mongodb-query-connector/src/value.rs

```diff
@@ -80,6 +80,20 @@ fn convert_composite_object(cf: &CompositeFieldRef, pairs: Vec<(String, PrismaVa
     Ok(Bson::Document(doc))
 }
 
+impl IntoBson for (&ScalarFieldRef, PrismaValue, bool) {
```

Comment:
I feel like this isn't really a problem for the IntoBson trait to solve, but a function that you call at the right places. The BSON transformations are supposed to be more general, while this is a very specific use-case.